test/unit-test/ice_api/ice_api_utest.c

```diff
@@ -19,13 +87,273 @@
+    TEST_ASSERT_EQUAL( transactionIdStore,
+                       context.pStunBindingRequestTransactionIdStore );
+    TEST_ASSERT_EQUAL( testRandomFxn,
+                       context.cryptoFunctions.randomFxn );
+    TEST_ASSERT_EQUAL( testCrc32Fxn,
+                       context.cryptoFunctions.crc32Fxn );
+    TEST_ASSERT_EQUAL( testHmacFxn,
+                       context.cryptoFunctions.hmacFxn );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Add Host Candidate fail functionality for Bad Parameters.
+ */
+void test_iceAddHostCandidate_BadParams( void )
+{
+    IceContext_t context;
+    IceEndpoint_t endPoint = { 0 };
+    IceResult_t result;
+
+    result = Ice_AddHostCandidate( NULL,
+                                   &( endPoint ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+
+    result = Ice_AddHostCandidate( &( context ),
+                                   NULL );
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_BAD_PARAM,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Add Host Candidate fail functionality for Max Candidate Threshold.
+ */
+void test_iceAddHostCandidate_MaxCandidateThreshold( void )
+{
+    IceContext_t context;
+    IceEndpoint_t endPoint = { 0 };
+    IceResult_t result;
+
+    context.numLocalCandidates = 1000;
+    context.maxLocalCandidates = 1000;
+
+    result = Ice_AddHostCandidate( &( context ),
+                                   &( endPoint ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_MAX_CANDIDATE_THRESHOLD,
+                       result );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Validate ICE Add Host Candidate functionality.
+ */
+void test_iceAddHostCandidate( void )
+{
+    IceContext_t context = { 0 };
+    IceCandidate_t localCandidates[5];
+    IceEndpoint_t endPoint = { 0 };
+    IceResult_t result;
+
+    context.pLocalCandidates = &( localCandidates[ 0 ] );
+    context.maxLocalCandidates = 5;
+    context.numLocalCandidates = 0;
+
+    endPoint.isPointToPoint = 1;
+    const uint8_t * ipAddress = ( uint8_t * ) "192.168.1.100";
+    endPoint.transportAddress.family = 0;
+    memcpy( endPoint.transportAddress.address,
+            ipAddress,
+            16 );
+    // endPoint.transportAddress.address = inet_addr(ipAddress);
+    endPoint.transportAddress.port = 8080;
+
+    result = Ice_AddHostCandidate( &( context ),
+                                   &( endPoint ));
+
+    TEST_ASSERT_EQUAL( ICE_RESULT_OK,
+                       result );
+    TEST_ASSERT_EQUAL( 1,
+                       context.numLocalCandidates );
+    TEST_ASSERT_EQUAL( ICE_CANDIDATE_TYPE_HOST,
+                       context.pLocalCandidates[0].candidateType );
+    TEST_ASSERT_EQUAL( 0,
+                       context.pLocalCandidates[0].isRemote );
+    TEST_ASSERT_EQUAL( 2113929471,
```

Comment:
1 suggestion , while using this pre-calculated number - its always better to define a macro for this number like `HOST_CANDIDATE_PRIORITY` 
We can do something like this 
```
/*
 * The priority is calculated for a host candidate where pCandidate.isPointToPoint = 1.
 */
#define   HOST_CANDIDATE_PRIORITY             2113929471